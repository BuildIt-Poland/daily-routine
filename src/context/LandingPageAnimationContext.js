import React, { createContext, useState } from 'react';
import PropTypes from 'prop-types';

const LandingPageAnimationContext = createContext();
const { Provider } = LandingPageAnimationContext;

function LandingPageAnimationProvider({ children, history }) {
  const REDIRECT_DELAY = 1000;

  function redirectWithDelay(url) {
    setTimeout(() => {
      history.push(url);
    }, REDIRECT_DELAY);
  }

  function animateAndRedirect(redirectUrl) {
    toggleAnimation(state => ({
      ...state,
      isAnimated: true
    }));
    redirectWithDelay(redirectUrl);
  }

  function stopAnimation() {
    toggleAnimation(state => ({
      ...state,
      isAnimated: false
    }));
  }

  const initialState = {
    isAnimated: false,
    animateAndRedirect,
    stopAnimation
  };

  const [animation, toggleAnimation] = useState(initialState);

  return <Provider value={animation}>{children}</Provider>;
}

LandingPageAnimationProvider.propTypes = {
  children: PropTypes.node,
  history: PropTypes.object
};

export { LandingPageAnimationContext, LandingPageAnimationProvider };
