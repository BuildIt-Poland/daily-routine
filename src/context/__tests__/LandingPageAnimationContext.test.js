import React from 'react';
import { create } from 'react-test-renderer';

import { LandingPageAnimationProvider } from '../LandingPageAnimationContext';

describe('COMPONENT - LandingPageAnimationProvider', () => {
  it('renders correctly', () => {
    const component = create(<LandingPageAnimationProvider></LandingPageAnimationProvider>);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
