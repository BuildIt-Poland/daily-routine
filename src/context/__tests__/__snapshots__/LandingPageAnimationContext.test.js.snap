// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - LandingPageAnimationProvider renders correctly 1`] = `null`;
